tbpore/external_tools.py

```diff
@@ -0,0 +1,33 @@
+import subprocess
+from typing import List
+import shlex
+import os
+from loguru import logger
+from pathlib import Path
+import hashlib
+
+
+class ExternalTool:
+    def __init__(self, tool: str, input: str, output: str, params: str, logdir: Path = Path("logs")):
+        self.command: List[str] = self._build_command(tool, input, output, params)
+        os.makedirs(logdir, exist_ok=True)
+        command_hash = hashlib.sha256(" ".join(self.command).encode("utf-8")).hexdigest()
+        logfile_prefix: Path = logdir / f"{tool}_{command_hash}"
+        self.out_log = f"{logfile_prefix}.out"
+        self.err_log = f"{logfile_prefix}.err"
+
+    @property
+    def command_as_str(self) -> str:
+        return " ".join(self.command)
```

Comment:
I wonder if https://docs.python.org/3.8/library/shlex.html#shlex.join might be safer?

Reply:
Nice spot, will do that, even though `command_as_str` is never executed, is just printed to log. So it should be safe anyway, but I think `shlex.join` will make it even safer

Reply:
Done in https://github.com/mbhall88/tbpore/pull/8/commits/39d4cb4c8e226f14e044d8d5a96650d2aabc522f